core/types/transaction.go

```diff
@@ -189,23 +189,13 @@ func (tx *Transaction) setDecoded(inner TxData, size int) {
 }
 
 func sanityCheckSignature(v *big.Int, r *big.Int, s *big.Int, maybeProtected bool) error {
-	if isProtectedV(v) && !maybeProtected {
```

Comment:
Since we are now only allowing protected transactions, I wonder if it would make more sense to return an error here if `!isProtected(v)`

Reply:
Fixed